rust-runtime/aws-smithy-http/src/body.rs

```diff
@@ -49,27 +50,40 @@ impl Debug for SdkBody {
     }
 }
 
-type BoxBody = http_body::combinators::BoxBody<Bytes, Error>;
-
-#[pin_project(project = InnerProj)]
-enum Inner {
-    Once(#[pin] Option<Bytes>),
-    Streaming(#[pin] hyper::Body),
-    Dyn(#[pin] BoxBody),
-
-    /// When a streaming body is transferred out to a stream parser, the body is replaced with
-    /// `Taken`. This will return an Error when polled. Attempting to read data out of a `Taken`
-    /// Body is a bug.
-    Taken,
+pub type BoxBody = http_body::combinators::BoxBody<Bytes, Error>;
+
+pin_project! {
+    #[project = InnerProj]
+    enum Inner {
+        Once {
+            #[pin]
```

Comment:
Do `Once` and `Dyn` actually need `#[pin]`? I understand why `Streaming` needs it, but I'm fuzzy on why these would need it.

Reply:
`Dyn` complains without it but `Once` seems to be ok when I remove it